### ISSUE TITLE
Add support for priority

### ISSUE THREAD

**Issue by poshcodebear** [poshcodebear/PoshMailKit#10]
Parameter -Priority, takes the following options:

- High
- Low
- Normal

This is actually an enum of type Net.Mail.MailPriority; it could probably be used directly, unless there's a concern of it being deprecated or removed in the future (or if MailKit has its own implementation of it)

**Comment by poshcodebear** [comment]
As MimeKit has an enum for this, `MimeKit.MessagePriority`, I've decided to use that; however, it looks like it uses different names than `System.Net.Meail.MailPriority` does, with High and Low being replaced with Urgent and NonUrgent (respectively).  This corresponds correctly with the Priority field set in the header, so it is in that sense the _better_ enum for this.

At this time, for proper backwards compatibility with Send-MailMessage, I'm adding a MailPriority enum to the project directly that uses the same names and values and having the Priority parameter take that; however, I'm having to translate it into the `MimeKit.MessagePriority` enum before attaching it to the message.

As this is also intended to be a forward-looking cmdlet, I'm also adding a MessagePriority parameter which will take the proper type directly and setting the two parameters up into their own mutually exclusive parameter sets, Legacy and Modern.  This may not be a good solution long term as we run into other cases where a legacy method doesn't make sense in the modern context and will need to be translated with a modern counterpart, and if and when we run into those cases, we'll have to reevaluate whether we want to lock users in to only using legacy or modern parameters, or if we want them to be able to mix and match otherwise.

One issue I can think of is if a modern method is added alongside a legacy counterpart that was not in the legacy set prior, using the either/or methodology would cause any use cases where someone was using that parameter and any of the modern parameters to break.

Anyway, short version is I think this won't be a long-term solution to the problem and I might have to do some rather hideous parameter set juggling to only make counterparts mutually exclusive to each other.
